providers/azure/dns.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2020 The Terraformer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azure
+
+import (
+	"context"
+	"log"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/dns/mgmt/2018-05-01/dns"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/GoogleCloudPlatform/terraformer/terraformutils"
+	"github.com/hashicorp/go-azure-helpers/authentication"
+)
+
+type DNSGenerator struct {
+	AzureService
+}
+
+func (g *DNSGenerator) listRecordSets(resourceGroupName string, zoneName string, top *int32) ([]terraformutils.Resource, error) {
+	var resources []terraformutils.Resource
+	ctx := context.Background()
+	subscriptionID := g.Args["config"].(authentication.Config).SubscriptionID
+	RecordSetsClient := dns.NewRecordSetsClient(subscriptionID)
+	RecordSetsClient.Authorizer = g.Args["authorizer"].(autorest.Authorizer)
+
+	recordSetIterator, err := RecordSetsClient.ListAllByDNSZoneComplete(ctx, resourceGroupName, zoneName, top, "")
+	if err != nil {
+		return nil, err
+	}
+	for recordSetIterator.NotDone() {
+		recordSet := recordSetIterator.Value()
+		// NOTE:
+		// Format example: "Microsoft.Network/dnszones/AAAA"
+		recordType := *recordSet.Type
+		switch {
+		case strings.HasSuffix(recordType, "/A"):
```

Comment:
you can change to 
```
terraformResourceName = ""
switch {
case 'A':
terraformResourceName="fff"
case 'b':
terraformResourceName = "bbb"
}
resources = append(resources, terraformutils.NewSimpleResource(
				*recordSet.ID,
				*recordSet.Name,
				terraformResourceName,
				g.ProviderName,
				[]string{}))

}
```

less code

Reply:
updated to do simple check for support/not